### ISSUE TITLE
Dealing with de-registered Debian Cassandra package versions

### ISSUE THREAD

**Issue by zmarois** [voxpupuli/puppet-cassandra#421]
With Cassandra's move back to semantic versioning, they have begun introducing patch versions. However, the Apache Debian repo only maintains the latest patch version for each minor version in their packages list.

I [asked about this](https://www.mail-archive.com/user@cassandra.apache.org/msg55240.html) on the Users mailing list, to which the response was "This is a feature. Users really should be installing the latest release version.". I had made my case for why I'd want a specific version to their IRC channel also. So this is not a bug we can expect to be fixed, but an issue to work around if we want to install a specific version.

While this is not a direct problem caused by this puppet module, it is a concern for users of it if we utilize [package_ensure for a specific version](https://github.com/voxpupuli/puppet-cassandra/blob/master/manifests/init.pp#L59).

I'd like to understand if the maintainers of this project have an opinion on dealing with this. I see the following options:
* Build a different mechanism to install a specific Cassandra version
* Build any safeguards to at least inform users (API documentation, making package_ensure invalid to set to a specific version > 3.11 on debian systems) that specific versions are only supported if it happens to be the latest version
* Just let the apt error about missing specific version bubble to the user when Apache releases a new patch (current implementation)

I'm uncomfortable installing the latest version always, as I'd like to keep my cluster all running the same version, and if I need to replace an unhealthy node, I don't want to have to upgrade the rest of my cluster first. So I'm probably going to build some mechanism to support that in my system.

**Comment by dallinb** [comment]
I agree that install `latest` is probably the last option anyone should adopt from a configuration management point of view. Could you expand upon your first suggested option?

**Comment by zmarois** [comment]
Either hack together some way of constructing a local repo that points to the .deb file in the archive (note that they [do all still exist](http://dl.bintray.com/apache/cassandra/pool/main/c/cassandra/), they just aren't in the Package list), or, when a specific version is specified >= 3.11, [use dpkg](https://puppet.com/docs/puppet/5.4/type.html#package-provider-dpkg). Perhaps not allow a specific version >= 3.11 and require a full url to the package.
